span.go

```diff
@@ -135,15 +133,38 @@ type FinishOptions struct {
 	// (per StartSpanOptions).
 	FinishTime time.Time
 
-	// BulkLogData allows the caller to specify the contents of many Log()
+	// LogRecords allows the caller to specify the contents of many LogFields()
 	// calls with a single slice. May be nil.
 	//
-	// None of the LogData.Timestamp values may be .IsZero() (i.e., they must
+	// None of the LogRecord.Timestamp values may be .IsZero() (i.e., they must
 	// be set explicitly). Also, they must be >= the Span's start timestamp and
 	// <= the FinishTime (or time.Now() if FinishTime.IsZero()). Otherwise the
 	// behavior of FinishWithOptions() is undefined.
 	//
-	// If specified, the caller hands off ownership of BulkLogData at
+	// If specified, the caller hands off ownership of LogRecords at
 	// FinishWithOptions() invocation time.
+	//
+	// If specified, the (deprecated) BulkLogData must be nil or empty.
+	LogRecords []LogRecord
+
+	// BulkLogData is DEPRECATED.
 	BulkLogData []LogData
 }
+
+// LogData is DEPRECATED
+type LogData struct {
+	Timestamp time.Time
+	Event     string
+	Payload   interface{}
+}
+
+// ToLogRecord converts a deprecated LogData to a non-deprecated LogRecord
+func (ld *LogData) ToLogRecord() LogRecord {
```

Comment:
@yurishkuro I did do an informal benchmark before writing my comment above (just to sanity-check things). In this case it's a wash. Just FYI, here are some more formal numbers comparing a thin (but not empty) struct and a thick (1K) struct...

Here's `bench_test.go`:

```go
package value_ptr

import "testing"

/////////////////////////////////////////
// THIN STRUCT (8 bytes)
/////////////////////////////////////////
type Thin struct {
       	idx int
       	data [4]byte
}

func (t Thin) ByValue() {
       	t.data[t.idx]++
       	t.idx++
       	if t.idx > len(t.data) { t.idx = 0 }
}

func (t *Thin) ByPointer() {
       	t.data[t.idx]++
       	t.idx++
       	if t.idx >= len(t.data) { t.idx = 0 }
}

func BenchmarkThinByValue(b *testing.B) {
       	t := Thin{idx: 0}
       	for i := 0; i < b.N; i++ {
       		t.ByValue()
       	}
}

func BenchmarkThinByPointer(b *testing.B) {
       	t := Thin{idx: 0}
       	for i := 0; i < b.N; i++ {
       		t.ByPointer()
       	}
}

/////////////////////////////////////////
// THICK STRUCT (1K)
/////////////////////////////////////////
type Thick struct {
       	idx int
       	data [1020]byte
}

func (t Thick) ByValue() {
       	t.data[t.idx]++
       	t.idx++
       	if t.idx > len(t.data) { t.idx = 0 }
}

func (t *Thick) ByPointer() {
       	t.data[t.idx]++
       	t.idx++
       	if t.idx >= len(t.data) { t.idx = 0 }
}

func BenchmarkThickByValue(b *testing.B) {
       	t := Thick{idx: 0}
       	for i := 0; i < b.N; i++ {
       		t.ByValue()
       	}
}

func BenchmarkThickByPointer(b *testing.B) {
       	t := Thick{idx: 0}
       	for i := 0; i < b.N; i++ {
       		t.ByPointer()
       	}
}
bhs@bhs-ls:~/tmp/go/src/value_ptr$
```

... and here are the results:

```
BenchmarkThinByValue-8         	1000000000     	         2.64 ns/op
BenchmarkThinByPointer-8       	500000000      	         3.12 ns/op
BenchmarkThickByValue-8        	50000000       	        30.0 ns/op
BenchmarkThickByPointer-8      	500000000      	         3.40 ns/op
```